frontend/components/LSPAssets.tsx

```diff
@@ -0,0 +1,195 @@
+import { WalletContext } from '@/components/wallet/WalletContext';
+import ERC725, { ERC725JSONSchema } from '@erc725/erc725.js';
+import LSP3ProfileSchema from '@erc725/erc725.js/schemas/LSP3ProfileMetadata.json';
+import LSP8IdentifiableDigitalAsset from '@lukso/lsp-smart-contracts/artifacts/LSP8IdentifiableDigitalAsset.json';
+import { detectLSP, LSPType, TokenInfo } from '@/utils/tokenUtils';
+import LSP7Panel from '@/components/LSP7Panel';
+import LSP8Panel from '@/components/LSP8Panel';
+import { constants } from '@/app/constants';
+
+export default function LSPAssets() {
+  const walletContext = useContext(WalletContext);
+  const [loading, setLoading] = useState(true);
+  const [lsp7Assets, setLsp7Assets] = useState<TokenInfo[]>([]);
+  const [lsp8Assets, setLsp8Assets] = useState<TokenInfo[]>([]);
+  const toast = useToast();
+
+  if (!walletContext) {
+    throw new Error('WalletConnector must be used within a WalletProvider.');
+  }
+
+  const { account, graveVault } = walletContext;
+
+  /**
+   * Fetch assets from the grave vault.
+   * This function is called when the page loads and when an asset is revived
+   */
+  const fetchAssets = useCallback(async () => {
+    if (!graveVault || graveVault === constants.ZERO_ADDRESS) {
+      setLoading(false);
+      return;
+    }
+    setLoading(true);
+    const erc725js = new ERC725(
+      LSP3ProfileSchema as ERC725JSONSchema[],
+      graveVault,
+      window.lukso,
+      {
+        ipfsGateway: constants.IPFS,
+      }
+    );
+
+    try {
+      const receivedAssetsResults = await erc725js.fetchData(
+        'LSP5ReceivedAssets[]'
+      );
+      const lsp7Results: TokenInfo[] = [];
+      const lsp8Results: TokenInfo[] = [];
+      const detectAssetCalls: Promise<TokenInfo | undefined>[] = [];
+
+      for (const assetAddress of receivedAssetsResults.value as string[]) {
+        detectAssetCalls.push(
+          detectLSP(assetAddress, graveVault, LSPType.LSP7DigitalAsset)
+        );
+        detectAssetCalls.push(
+          detectLSP(
+            assetAddress,
+            graveVault,
+            LSPType.LSP8IdentifiableDigitalAsset
+          )
+        );
+      }
+
+      const receivedAssetsWithTypes = await Promise.all(detectAssetCalls);
+      for (const asset of receivedAssetsWithTypes) {
+        if (!asset) continue;
+
+        if (asset.type === LSPType.LSP7DigitalAsset) {
+          lsp7Results.push(asset);
+        } else if (asset.type === LSPType.LSP8IdentifiableDigitalAsset) {
+          const contract = new ethers.Contract(
+            asset.address as string,
+            LSP8IdentifiableDigitalAsset.abi,
+            new ethers.providers.Web3Provider(window.lukso)
+          );
+          const tokenIds = await contract.tokenIdsOf(graveVault);
+          tokenIds.forEach((tokenId: string) => {
+            lsp8Results.push({
+              ...asset,
+              tokenId: tokenId.toString(),
+            });
+          });
+        }
+      }
```

Comment:
should we extract this to a function

Reply:
don't think it makes too much sense given how specific it is to this useCallback logic. if this logic gets more complex we should just move it out into its own `useFetchAssets` function and import it here.